alica_engine/src/engine/AlicaEngine.cpp

```diff
@@ -31,15 +31,15 @@ void AlicaEngine::abort(const std::string& msg)
 /**
  * The main class.
  */
-AlicaEngine::AlicaEngine(AlicaContext& ctx, const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine, const essentials::IdentifierConstPtr agentID)
+AlicaEngine::AlicaEngine(AlicaContext& ctx, const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine, const essentials::Identifier& agentID)
         : _ctx(ctx)
         , _stepCalled(false)
         , _stepEngine(stepEngine)
         , _log(this)
         , _modelManager(_planRepository)
         , _masterPlan(_modelManager.loadPlanTree(masterPlanName))
         , _roleSet(_modelManager.loadRoleSet(roleSetName))
-        , _teamManager(this, (static_cast<bool>(agentID) ? getIDFromBytes(agentID->toByteVector().data(), agentID->toByteVector().size()) : nullptr))
+        , _teamManager(this, (static_cast<bool>(agentID) ? getIDFromBytes(agentID.toByteVector().data(), agentID.toByteVector().size()) : nullptr))
```

Comment:
I guess, this will always evaluate to "true" in case of the non-sense representing / default initialised Identifier.

Reply:
nope. bool operator is overloaded in identifier like this:
operator bool() const { return !_id.empty(); }